code/modules/mob/mob_movement.dm

```diff
@@ -157,6 +157,8 @@
 						M.stop_pulling()
 
 		switch(mob.m_intent)
+			if("sprint")
+				move_delay += config.run_speed

```

Comment:
the more damage a person has, the less they can sprint before landing on their face. If they have the damage slowdown it gets comically short